src/firefox/index.js

```diff
@@ -317,13 +317,48 @@ export function configureProfile(
   return Promise.resolve(profile);
 }
 
+export type getProfileFN = (profileName: string) => Promise<string | void>;
+
+export type CreateProfileFinderParams = {|
+  userDirectoryPath?: string,
+  FxProfile?: typeof FirefoxProfile
+|}
+
+export function defaultCreateProfileFinder(
+  {
+    userDirectoryPath,
+    FxProfile = FirefoxProfile,
+  }: CreateProfileFinderParams = {}
+): getProfileFN {
+  const finder = new FxProfile.Finder(userDirectoryPath);
+  const readProfiles = promisify(finder.readProfiles, finder);
+  const getPath = promisify(finder.getPath, finder);
+  return async (profileName: string): Promise<string | void> => {
+    try {
+      await readProfiles();
+      const hasProfileName = finder.profiles.filter(
+        (profileDef) => profileDef.Name === profileName).length !== 0;
+      if (hasProfileName) {
+        return await getPath(profileName);
+      }
+    } catch (error) {
+      if (isErrorWithCode('ENOENT', error)) {
```

Comment:
Do you mind to reverse this if and remove the else block?
```js
if (!isErrorWithCode('ENOENT', error)) {
  throw error;
}
log.warn('...');
```

Also, I'm thinking that we should tweak the warning message a bit, e.g.  
```js
log.warn('Unable to find Firefox profiles.ini');
```